projects/public-search/src/app/patron-profile/patron-profile-personal/patron-profile-personal.component.ts

```diff
@@ -29,4 +29,56 @@ export class PatronProfilePersonalComponent {
 
   /** patron record */
   @Input() patron: IPatron;
+
+  /** Show or hide user data */
+  hideUserData = false;
+
+  /** Show or hide user data form editor */
+  hideDataEditor = true;
+
+  /** Show or hide change password editor */
+  hidePassword = true;
+
+  /**
+   * Check if the user profile is on read only
+   *
+   * @returns boolean
```

Comment:
A general remark here: the `returns` comment contains only the type which is already in the method definition. It would be more informative to describe the content of the return value, such as 'true if... false else` or for other methods for example `the list of modified items` etc.